cowprotocol/services
# Fetch `Competition` from database

# Description
For a given auction_id, implements fetching a related auction object from and competition solution from database. These  objects should be the last piece of information needed to give the full context to a settlement, enough to do whatever is needed in the autopilot domain.

# Changes


- [ ] Implemented `infra::persistence::get_competition` to get auction object and competition solution from database
- [ ] fetched objects become a part of a `domain::Settlement`

## How to test
Added test log to `OnSettlementEventUpdater` that reports differences in the promised score vs delivered score. This should be enough to observe different settlements in prod. Locally, I've observed the output of different e2e tests and it looks good.

Non-failure of any e2e tests should give a sanity check.
